### PR TITLE
feat: form components

### PR DESCRIPTION
**Pull request by Robbert**
# Radio button

Voorzetje voor de Radio button component (backlog: #33 en nl-design-system/backlog#65)

## Terminologie

- **radio button**: "Radio button" heeft [een Wikipedia-lemma](https://en.wikipedia.org/wiki/Radio_button), [WAI-ARIA Authoring Practices 1.2](https://www.w3.org/TR/wai-aria-practices-1.2/#radiobutton) noemt het ook "radio button" en "radio group".
- **checked**: [WAI-ARIA Authoring Practices 1.2](https://www.w3.org/TR/wai-aria-practices-1.2/#radiobutton) en HTML noemen de staat van de radio button "checked", CSS heeft de `:checked` pseudo-class.

## Class names

- `.utrecht-radio-button`

## Design tokens

Nog geen design tokens.

---

# Button

Voorzetje voor Button component (backlog: #10 en nl-design-system/backlog#38).

## Terminologie

- **button**: van het `<button>` HTML element, `role="button"` in ARIA.

## States

- `hover`
- `focus`
- `disabled` (TODO)

## Class names

- `.utrecht-button`
- `.utrecht-button--focus`
- `.utrecht-button--hover`

## Design tokens

- Button
  - `--utrecht-button-border-width`
  - `--utrecht-button-focus-transform-scale`
  - `--utrecht-button-font-size`
  - `--utrecht-button-margin-block`
  - `--utrecht-button-margin-inline`
  - `--utrecht-button-padding-block`
  - `--utrecht-button-padding-inline`
  - Modifier: primary action
    - `--utrecht-button-primary-action-background-color`
    - `--utrecht-button-primary-action-color`
    - `--utrecht-button-primary-action-hover-background-color`

---

# Checkbox

Voorzetje voor Checkbox component (backlog: #32 en nl-design-system/backlog#64)

## Terminologie

- **checkbox**: `type="checkbox"` in HTML, [`role="checkbox"`](https://www.w3.org/TR/wai-aria-1.2/#checkbox) in WAI-ARIA 1.2, "checkbox" in [WAI-ARIA Authoring Practices](https://www.w3.org/TR/wai-aria-practices-1.2/#checkbox).
- **checked**: `checked` attribuut in HTML, `aria-checked="true"` in [WAI-ARIA](https://www.w3.org/TR/wai-aria-1.2/#aria-checked), `:checked` pseudo-class in CSS.

## States

- checked

## Class names

- `utrecht-checkbox`

## Design tokens

Nog geen.

---

# Text box

Voorzetje voor Text Box component (backlog: #30 en nl-design-system/backlog#5)

## Terminologie

- **textbox**: [`role="textbox"` in WAI-ARIA](https://www.w3.org/TR/wai-aria-1.2/#textbox), [Text box lemma op Wikipedia](https://en.wikipedia.org/wiki/Text_box). HTML noemt het "text control" en text edit control". [MDN noemt het](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/text) "text field" en "text input".
- **invalid**: ...
- **disabled**: ...
- **read-only**: ...

## States

- invalid
- disabled
- read-only

## Class names

- `.utrecht-textbox`
- `.utrecht-textbox--invalid`
- `.utrecht-textbox--disabled`
- `.utrecht-textbox--read-only`

## Design Tokens

- Textbox:
  - `--utrecht-textbox-border-bottom-width`
  - `--utrecht-textbox-border-color`
  - `--utrecht-textbox-border-radius`
  - `--utrecht-textbox-border-width`
  - `--utrecht-textbox-color`
  - `--utrecht-textbox-font-family`
  - `--utrecht-textbox-font-size`
  - `--utrecht-textbox-max-width`
  - `--utrecht-textbox-padding-block-end`
  - `--utrecht-textbox-padding-block-start`
  - `--utrecht-textbox-padding-inline-end`
  - `--utrecht-textbox-padding-inline-start`
  - Modifier: `disabled`
    - `--utrecht-textbox-disabled-border-color`
    - `--utrecht-textbox-disabled-color`
  - Modifier: `invalid`
    - `--utrecht-textbox-invalid-border-color`
    - `--utrecht-textbox-invalid-border-width`
  - Modifier: `read-only`:
    - `--utrecht-textbox-read-only-border-color`
    - `--utrecht-textbox-read-only-color`

---

# Textarea

Voorzetje voor Textarea component (backlog: #36 en nl-design-system/backlog#40)

## Terminologie

- **textarea**: van het `<textarea>` element in HTML.

## States

- invalid
- disabled
- read-only

## Class names

- `.utrecht-textarea`
- `.utrecht-textarea--invalid`
- `.utrecht-textarea--disabled`
- `.utrecht-textarea--read-only`

## Design tokens

- Textarea:
  - `--utrecht-textarea-border-bottom-width`
  - `--utrecht-textarea-border-color`
  - `--utrecht-textarea-border-radius`
  - `--utrecht-textarea-border-width`
  - `--utrecht-textarea-color`
  - `--utrecht-textarea-font-family`
  - `--utrecht-textarea-font-size`
  - `--utrecht-textarea-max-width`
  - `--utrecht-textarea-padding-block-end`
  - `--utrecht-textarea-padding-block-start`
  - `--utrecht-textarea-padding-inline-end`
  - `--utrecht-textarea-padding-inline-start`
  - Modifier: `disabled`
    - `--utrecht-textarea-disabled-border-color`
    - `--utrecht-textarea-disabled-color`
  - Modifier: `invalid`
    - `--utrecht-textarea-invalid-border-color`
    - `--utrecht-textarea-invalid-border-width`
  - Modifier: `read-only`:
    - `--utrecht-textarea-read-only-border-color`
    - `--utrecht-textarea-read-only-color`

---

# Radio button group

Voorzetje voor de Radio button group (backlog: #51)

---

# Checkbox group

Voorzetje voor de Checkbox group (backlog: #50)

### CODE REVIEW

File: src/button/style.css [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+  border-width: var(--utrecht-button-border-width);
+  padding-inline-start: var(--utrecht-button-padding-inline);
+  padding-inline-end: var(--utrecht-button-padding-inline);
+  padding-block-start: var(--utrecht-button-padding-block);
+  padding-block-end: var(--utrecht-button-padding-block);
+
+  margin-inline-start: var(--utrecht-button-margin-inline);
+  margin-inline-end: var(--utrecht-button-margin-inline);
+  margin-block-start: var(--utrecht-button-margin-block);
+  margin-block-end: var(--utrecht-button-margin-block);
```
Comment (Yolijn) [inline src/button/style.css:20]: let's move the margins to a `--distanced` variant

File: src/checkbox/index.css [hunk 2]
```diff
@@ -0,0 +1,11 @@
+/**
+ * @license EUPL-1.2
+ * Copyright (c) 2021 Robbert Broersma
+ */
+
+.utrecht-checkbox {
+  margin-block-start: 0;
```
Comment (Yolijn) [inline src/checkbox/index.css:7]: #moveTheMargins

File: src/form-field-checkbox/index.css [hunk 3]
```diff
@@ -0,0 +1,14 @@
+/**
+ * @license EUPL-1.2
+ * Copyright (c) 2021 Robbert Broersma
+ */
+
+.utrecht-form-field-checkbox {
+  font-family: var(--utrecht-document-font-family, inherit);
+  margin-block-start: var(--utrecht-form-field-margin-block-start, var(--utrecht-paragraph-margin-block-start));
```
Comment (Yolijn) [inline src/form-field-checkbox/index.css:8]: #moveTheMargins

File: src/form-field-radio/index.css [hunk 4]
```diff
@@ -0,0 +1,15 @@
+/**
+ * @license EUPL-1.2
+ * Copyright (c) 2021 Robbert Broersma
+ */
+
+.utrecht-form-field-radio {
+  font-size: var(--utrecht-paragraph-font-size);
+  font-family: var(--utrecht-document-font-family, inherit);
+  margin-block-start: var(--utrecht-form-field-margin-block-start, var(--utrecht-paragraph-margin-block-start));
```
Comment (Yolijn) [inline src/form-field-radio/index.css:9]: #moveTheMargins

File: src/radio-button/index.css [hunk 5]
```diff
@@ -0,0 +1,11 @@
+/**
+ * @license EUPL-1.2
+ * Copyright (c) 2021 Gemeente Utrecht
+ */
+
+.utrecht-radio-button {
+  margin-block-start: 0;
```
Comment (Yolijn) [inline src/radio-button/index.css:7]: #moveTheMargins

**Review by Yolijn — CHANGES_REQUESTED** [review]

File: src/checkbox/index.css (hunk 2 above)
Comment (Robbert) [reply src/checkbox/index.css:7]: This is to reset a native margin of 3px, I'll add a comment to clarify.

File: src/radio-button/index.css (hunk 5 above)
Comment (Robbert) [reply src/radio-button/index.css:7]: This is to reset a native margin of 3px, I'll add a comment to clarify.

**Review by Yolijn — APPROVED** [review]

**Pull request merged**